scripts/compile_proto.py

```diff
@@ -0,0 +1,39 @@
+"""
+Generate the Python code for the gRPC proto files.
+"""
+
+import subprocess
+from pathlib import Path
+
+# repository root directory
+_repo_root = Path(__file__).parent.parent
+
+
+def main():
```

Comment:
this script generates the grpc code.

```shell
python scripts/compile_proto.py
```

i left the paths like they currently are - can revisit that later if desired.